src/main/java/dev/espi/ebackup/eBackup.java

```diff
@@ -37,6 +37,7 @@
 
     // lock
     AtomicBoolean isInBackup = new AtomicBoolean(false);
+    AtomicBoolean isInUpload = new AtomicBoolean(false);
```

Comment:
I don't think it's a good idea to lock to only allow one upload at a time, if a backup was created later, then let the later backup also get uploaded.